src/Octoshift/GithubApi.cs

```diff
@@ -105,6 +106,23 @@ public virtual async Task RemoveTeamMember(string org, string teamSlug, string m
             await _retryPolicy.HttpRetry(() => _client.DeleteAsync(url), _ => true);
         }
 
+        public virtual async Task<bool> DoesRepoExist(string org, string repo)
+        {
+            var url = $"{_apiUrl}/repos/{org}/{repo}";
+            try
+            {
+                await _client.GetNonSuccessAsync(url, HttpStatusCode.NotFound);
+            }
+            catch (HttpRequestException ex)
+            {
+                if (ex.StatusCode == HttpStatusCode.OK)
+                {
+                    return true;
+                }
+            }
+            return false;
```

Comment:
1. It's a bit strange to always rely on an exception to determine the result of a method, instead I would naturally return either `false` or `true` and then check only the exceptional case in the `catch` block:
```c#
try
{
    await _client.GetAsync(url);
    return true;
}
catch (HttpRequestException ex) when (ex.StatusCode == HttpStatusCode.NotFound)
{
    return false;
}
```
2. The catch block actually swallows all `HttpRequestExceptions`! so for example if there is a rate limit exception it swallows it and returns `false` instead. The way I re-wrote it above won't have this problem because the catch block is only going to get executed if the condition is true.

Reply:
we originally had the code very similar to this, the problem is in the normal case (repo does not exist) the GetAsync will retry 5 times, which will affect perf.

Reply:
then I will reverse the logic like this. We still shouldn't swallow all exceptions:

```c#
try
{
    await _client. GetNonSuccessAsync(url, HttpStatusCode.NotFound);
    return false;
}
catch (HttpRequestException ex) when (ex.StatusCode == HttpStatusCode.OK)
{
    return true;
}
```

Reply:
Addressed in https://github.com/github/gh-gei/pull/787/commits/2afeca3056b0bd970e90cee0a09daedb5ec0c9c5! 🎉